modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/jest.config.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+// eslint-disable-next-line no-undef
+module.exports = {
+	collectCoverageFrom: [
+		'**/*.{js,jsx}',
+		'!**/node_modules/**',
+		'!**/vendor/**',
+	],
+	moduleFileExtensions: ['js', 'jsx', 'html'],
+	modulePathIgnorePatterns: [
+		'<rootDir>[/\\\\](build|docs|node_modules|deploy|scripts)[/\\\\]',
+	],
+	setupFilesAfterEnv: ['./setupTests.js'],
```

Comment:
When using `npm-scripts` some mocks for Liferay's [globals are added automatically](https://github.com/liferay/liferay-frontend-projects/blob/master/projects/npm-tools/packages/npm-scripts/src/jest/setup.js#L10). However, seeing `setupTests.js` file you're needing to add an enzyme setup there. Is there a way to add an override option on `npm-scripts.config.js` file to just extend our configs? @bryceosterhaus @kresimir-coko @izaera ?

Reply:
I believe that for our jest configs, we merge our config with any that is passed in. So for this example, the config used when running `test` with `@liferay/npm-scripts`. The first file comes from [here](https://github.com/liferay/liferay-frontend-projects/blob/ef22d1d1c4d76377138a924e72357266949ac881/projects/npm-tools/packages/npm-scripts/src/jest/setup.js)

```json
  "setupFiles": [
    "/Users/bryceosterhaus/repos/liferay-portal/modules/node_modules/@liferay/npm-scripts/src/jest/setup.js",
    "<rootDir>/setupTests.js"
  ]
```

Reply:
What @bryceosterhaus says. In addition, we have more or less decided in another conversation (that I remember but don't know where it is) that we wanted our tools to have as less config as possible, but leaving standard config files (those used by other industry tools) present in the project.

So, according to that rule, jest config that is not injected, should go in `jest.config.js`, not `npm-scripts.config.js`.

Am I right @bryceosterhaus ? :thinking:

Reply:
Yes, users should go for the recommended config for the tool. So for jest, we would recommend using `jest.config.js` and not putting it in the npm-scripts config.

By default, we use [cosmiconfig](https://github.com/davidtheclark/cosmiconfig) to search for the user configuration and it is a pretty standard tool for configs in js.